### PR TITLE
HPCC-29859 Publish accurate in-memory index sizes

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
See the results of the different searchindexes generated by the regression suite.  E.g.
Existing:
```
        Original Size | Size on Disk | %Compression | Memory Size
File     | 33.23 MB | 33.23 MB | 100.00% | 172.72 MB
Branches | 194.24 KB | 144.00 KB | 74.13% | 194.24 KB
Data     | 172.53 MB | 33.06 MB | 19.16% | 172.53 MB
```
New compression:

```
        Original Size | Size on Disk | %Compression | Memory Size
File     | 26.47 MB | 26.47 MB | 100.00% | 79.58 MB
Branches | 155.00 KB | 56.00 KB | 36.13% | 44.77 KB
Data     | 172.53 MB | 26.38 MB | 15.29% | 79.53 MB
```
Memory size is very different from original size for the new format

File: thorlcr/activities/indexwrite/thindexwriteslave.cpp [hunk 1]
```diff
@@ -241,6 +243,8 @@ class IndexWriteSlaveActivity : public ProcessSlaveActivity, public ILookAheadSt
                     numBranchNodes = builder->getNumBranchNodes();
                     numBlobNodes = builder->getNumBlobNodes();
                     offsetBranches = builder->getOffsetBranches();
+                    branchMemorySize = builder->getBranchMemorySize();
+                    leafMemorySize = builder->getLeafMemorySize();
```
Comment (jakesmith) [inline thorlcr/activities/indexwrite/thindexwriteslave.cpp:247]: not for now, but these stats should be collected/serialized in a more conventional way, whilst the query is running (in gatherActiveStats). Would need IKeyBuilder to expose its stats as CRuntimeStatisticCollection or similar.
I've opened a new JIRA: https://track.hpccsystems.com/browse/HPCC-29880

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good. Added 1 comment, but it's for a future improvement.

**Review by richardkchapman — APPROVED** [review]
I did not spot any issues

**Pull request merged**